dll/win32/comdlg32/filedlg.c

```diff
@@ -2826,8 +2826,98 @@ void FILEDLG95_OnOpenMessage(HWND hwnd, int idCaption, int idText)
   MessageBoxW(hwnd,strMsgText, strMsgTitle, MB_OK | MB_ICONHAND);
 }
 
+#ifdef __REACTOS__
+static LPWSTR FILEDLG95_GetFallbackExtension(FileOpenDlgInfos *fodInfos, LPWSTR lpstrPathAndFile)
+{
+    LPWSTR lpstrFilter, the_ext = NULL, pchDot = NULL;
+
+    /* Without lpstrDefExt, append no extension */
+    if (!fodInfos->defext)
+        return NULL;
+
+    /* Get filter extensions */
+    lpstrFilter = (LPWSTR) CBGetItemDataPtr(fodInfos->DlgInfos.hwndFileTypeCB,
+                                            fodInfos->ofnInfos->nFilterIndex - 1);
+    if (lpstrFilter != (LPWSTR)CB_ERR && lpstrFilter && *lpstrFilter)
+    {
+        LPWSTR pchSemicolon = wcschr(lpstrFilter, L';');
+
+        if (pchSemicolon)
+            *pchSemicolon = UNICODE_NULL;
+
+        pchDot = wcschr(lpstrFilter, L'.');
+
+        if (pchDot && pchDot[1] && !wcschr(pchDot, L'*') && !wcschr(pchDot, L'?'))
+            the_ext = StrDupW(pchDot + 1);
+
+        if (pchSemicolon)
+            *pchSemicolon = L';';
+    }
+
+    if (!the_ext && (!pchDot || pchDot[1]))
+    {
+        /* use default extension if no extension in filter */
+        the_ext = StrDupW(fodInfos->defext);
+    }
+
+    return the_ext; /* Needs LocalFree */
```

Comment:
Move this comment up before the function prototype (as part of its "documentation").

Reply:
Fixed.